targets/TARGET_Cypress/TARGET_PSOC6/psoc6csp/hal/src/cyhal_crc.c

```diff
@@ -36,19 +36,15 @@
 cy_rslt_t cyhal_crc_init(cyhal_crc_t *obj)
 {
     CY_ASSERT(NULL != obj);
-
-    memset(obj, 0, sizeof(cyhal_crc_t));
-    obj->resource.type = CYHAL_RSC_CRC;
-    return cyhal_crypto_reserve(&(obj->base), &(obj->resource));
+    return cyhal_crypto_reserve(&(obj->base), &(obj->resource), CYHAL_CRYPTO_CRC);
 }
 
 void cyhal_crc_free(cyhal_crc_t *obj)
```

Comment:
Should this be renamed to `cyhal_crypto_free()` now that it works with `CYHAL_RSC_CRYPTO`?

Reply:
cyhal_crc_free() performs additional checks for resource subtype before free it. So this function releases only CRC allocated object no other types

Reply:
OK